connectors/sync_job_runner.py

```diff
@@ -374,7 +376,9 @@ async def _sync_done(self, sync_status, sync_error=None):
 
         if await self.reload_connector():
             sync_cursor = (
-                self.data_provider.sync_cursor()
+                None
+                if not self.data_provider  # If we failed before initializing the data provider, we don't need to change the cursor
```

Comment:
Could the `None` sync_cursor cause problems down the line if there are intermittent issues with data provider initialization. 

I'm thinking of:
- we have connector, we run incremental syncs fine
- we once run into `if not self.data_provider` is `true` case
- we set this `None` cursor in connector doc `self.connector.sync_done(...`

How to recover for further incremental syncs (e.g. scheduled)? Can our logic communicate that user needs to run a full sync first given that the cursor is now `None`?

Reply:
Sync cursor only gets set if if the job completed successfully: 
https://github.com/elastic/connectors/blob/main/connectors/protocol/connectors.py#L763-L764 

so we shouldn't be at risk here. If `self.data_provider` is not set, we didn't complete successfully.